### ISSUE TITLE
Does not build on nixos

### ISSUE THREAD

**Issue by tbdcit** [ifreund/river#121]
I am struggling to get river to build on nixos. Using the following `shell.nix` as recommended in the wiki:

```
with import <nixpkgs> {};

stdenv.mkDerivation rec {
  name = "river-environment";

  buildInputs = with pkgs; [
    zig
    wayland
    wayland-protocols
    wlroots
    libxkbcommon
    libudev
    libevdev
    xorg.libX11
    pixman
    libGL
  ];
  nativeBuildInputs = with pkgs; [ pkgconfig ];
}
```

I receive the following error
```
> nix-shell ./shell.nix
> zig build -Drelease-safe=true --prefix ./out install
Semantic Analysis [1861/2857] ./build.zig:142:41: error: expected type '[]const u8', found '(enum literal)'
            .step = std.build.Step.init(.Custom, "Scan Protocols", builder.allocator, make),
                                        ^
./build.zig:237:41: error: expected type '[]const u8', found '(enum literal)'
            .step = std.build.Step.init(.Custom, "Generate man pages", builder.allocator, make),
```

Any ideas on how to fix this?

**Comment by ifreund** [comment]
Hello and thank you for your interest in river. As stated in the readme, river currently depends on the master branch of zig due to an issue that has been fixed since the 0.6.0 release. You can download a static tarball of the release here: https://ziglang.org/download/

With the proper zig version that error should go away, let me know if you run into any other issues!

**Comment by tbdcit** [comment]
Thanks for the quick reply! Sorry I missed that in the readme. I have downloaded a static tarball of the master branch of zig but I am still running into some issues when I try to build:

```
> ./zig-linux-x86_64-0.6.0+198d1438e/zig build -Drelease-safe=true --prefix ./out install./river/Output.zig:246:27: error: expected type 'u32', found 'std.os.WaitPidResult'    if (!std.os.WIFEXITED(status) or std.os.WEXITSTATUS(status) != 0)                          ^./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared herepub const WaitPidResult = struct {                          ^
./river/Output.zig:299:39: note: referenced here
    self.layoutExternal(layout_count) catch |err| {
                                      ^
./river/command/spawn.zig:59:27: error: expected type 'u32', found 'std.os.WaitPidResult'
    if (!std.os.WIFEXITED(status) or
                          ^
./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared here
pub const WaitPidResult = struct {
                          ^

> ./zig-linux-x86_64-0.6.0+198d1438e/zig version
0.6.0+198d1438e
```

Any ideas?

**Comment by aruhier** [comment]
> Thanks for the quick reply! Sorry I missed that in the readme. I have downloaded a static tarball of the master branch of zig but I am still running into some issues when I try to build:
> 
> ```
> > ./zig-linux-x86_64-0.6.0+198d1438e/zig build -Drelease-safe=true --prefix ./out install./river/Output.zig:246:27: error: expected type 'u32', found 'std.os.WaitPidResult'    if (!std.os.WIFEXITED(status) or std.os.WEXITSTATUS(status) != 0)                          ^./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared herepub const WaitPidResult = struct {                          ^
> ./river/Output.zig:299:39: note: referenced here
>     self.layoutExternal(layout_count) catch |err| {
>                                       ^
> ./river/command/spawn.zig:59:27: error: expected type 'u32', found 'std.os.WaitPidResult'
>     if (!std.os.WIFEXITED(status) or
>                           ^
> ./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared here
> pub const WaitPidResult = struct {
>                           ^
> 
> > ./zig-linux-x86_64-0.6.0+198d1438e/zig version
> 0.6.0+198d1438e
> ```
> 
> Any ideas?

With this version, it doesn't build for me neither. The zig master release from yesterday managed to compile it, however.

**Comment by tbdcit** [comment]
```.
> ./zig-linux-aarch64-0.6.0+3ab4d112e/zig version 
bash: ./zig-linux-aarch64-0.6.0+3ab4d112e/zig: cannot execute binary file: Exec format error
```
With yesterdays.... not sure what is happening here.

Could you send commit ref of zig for which this is working?

**Comment by aruhier** [comment]
> ```
> > ./zig-linux-aarch64-0.6.0+3ab4d112e/zig version 
> bash: ./zig-linux-aarch64-0.6.0+3ab4d112e/zig: cannot execute binary file: Exec format error
> ```
> 
> With yesterdays.... not sure what is happening here.
> 
> Could you send commit ref of zig for which this is working?

I think you took the wrong version, if I compare to your logs from yesterday .You took the aarch64, not x86_64 bits version.

**Comment by ifreund** [comment]
> Thanks for the quick reply! Sorry I missed that in the readme. I have downloaded a static tarball of the master branch of zig but I am still running into some issues when I try to build:
> 
> ```
> > ./zig-linux-x86_64-0.6.0+198d1438e/zig build -Drelease-safe=true --prefix ./out install./river/Output.zig:246:27: error: expected type 'u32', found 'std.os.WaitPidResult'    if (!std.os.WIFEXITED(status) or std.os.WEXITSTATUS(status) != 0)                          ^./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared herepub const WaitPidResult = struct {                          ^
> ./river/Output.zig:299:39: note: referenced here
>     self.layoutExternal(layout_count) catch |err| {
>                                       ^
> ./river/command/spawn.zig:59:27: error: expected type 'u32', found 'std.os.WaitPidResult'
>     if (!std.os.WIFEXITED(status) or
>                           ^
> ./zig-linux-x86_64-0.6.0+198d1438e/lib/zig/std/os.zig:3127:27: note: std.os.WaitPidResult declared here
> pub const WaitPidResult = struct {
>                           ^
> 
> > ./zig-linux-x86_64-0.6.0+198d1438e/zig version
> 0.6.0+198d1438e
> ```
> 
> Any ideas?

This was caused by a recent breaking change on master zig. I just pushed e179690a9c85de9bf53ff3165c9796dcfda3ce3a to fix it. River now builds fine for me with zig `0.6.0+198d1438e` which is currently the most recent build available from ziglang.org. 

Sorry that building river is such a pain currently, this should be resolved with zig 0.7.0.

**Comment by tbdcit** [comment]
Thanks for all the help. I can confirm that river is now building fine.
